### PR TITLE
DTFS2-5313 GEF only bank create application redirect

### PR DESCRIPTION
**Pull request by Zainzzkk**
# Changes made: 

* Added hasGefAccessOnly property to bank which can only make gef applications in mock-data-loader
* added maker and checker for new gef-only bank
* made a redirect on Create new button to redirect to mandatory criteria if gef only bank (href redirect on nunjucks)
* Added e2e test
* Removed unneccesary console.log and changed 'ACBS' string to constant for matching

### CODE REVIEW

File: e2e-tests/portal/cypress/integration/journeys/dashboard/dashboard-deals/dashboard-deals-create-deal.spec.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+      cy.url().should('eq', relative('/gef/mandatory-criteria'));
+    });
+
+    it('When clicking continue, it should lead to name your application', () => {
+      dashboard.mandatoryCriteriaYes().click();
+      dashboard.continueButton().click();
+      cy.url().should('eq', relative('/gef/name-application'));
+    });
+  });
+
```
Comment (ttbarnes) [inline e2e-tests/portal/cypress/integration/journeys/dashboard/dashboard-deals/dashboard-deals-create-deal.spec.js:29]: Can we add a test for manually navigating by URL to some BSS "create/start a deal" pages, as a GEF only bank?

File: e2e-tests/portal/cypress/integration/journeys/dashboard/dashboard-deals/dashboard-deals-create-deal.spec.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+  describe('Create application as all other banks', () => {
```
Comment (ttbarnes) [inline e2e-tests/portal/cypress/integration/journeys/dashboard/dashboard-deals/dashboard-deals-create-deal.spec.js:30]: Could we rename descriptions so it's clearer?

```
context('Create application as GEF-only bank and GEF-and-BSS bank', () => {
...


describe('Create application as GEF-and-BSS bank', () => {

```
Just "all other banks" isn't super clear

File: utils/mock-data-loader/banks/index.js [hunk 3]
```diff
@@ -20,6 +20,17 @@ const BANKS = [
     companiesHouseNo: '00014259',
     partyUrn: '00300342',
   },
+  {
+    id: '999',
+    name: 'GEF Only Bank',
+    mga: ['UKEF-GEF-only-amended-and-restated-MGA-dated-31-January-2020.pdf'],
+    emails: [
+      'checker4@ukexportfinance.gov.uk',
```
Comment (ttbarnes) [inline utils/mock-data-loader/banks/index.js:28]: Can we check with Caroline what email she'd like to use for this bank

File: utils/mock-data-loader/portal/users.js [hunk 4]
```diff
@@ -24,6 +25,26 @@ const USERS = [
...
+    timezone: 'Europe/London',
+    roles: ['maker'],
+    bank: UKEF_GEF_ONLY_BANK,
+  },
+  {
+    username: 'BANK3_GEF_CHECKER1',
+    password: 'AbC!2345',
+    firstname: 'Tony',
+    surname: 'Sheridan',
+    email: 'maker2@ukexportfinance.gov.uk',
```
Comment (ttbarnes) [inline utils/mock-data-loader/portal/users.js:43]: For both these users need to check what email Caroline would like to use

**Review by ttbarnes — CHANGES_REQUESTED** [review]

File: e2e-tests/portal/cypress/integration/journeys/dashboard/dashboard-deals/dashboard-deals-create-deal.spec.js (hunk 2 above)
Comment (Zainzzkk) [reply e2e-tests/portal/cypress/integration/journeys/dashboard/dashboard-deals/dashboard-deals-create-deal.spec.js:30]: done

File: utils/mock-data-loader/banks/index.js (hunk 3 above)
Comment (Zainzzkk) [reply utils/mock-data-loader/banks/index.js:28]: Caroline is fine with this email address

File: utils/mock-data-loader/portal/users.js (hunk 4 above)
Comment (Zainzzkk) [reply utils/mock-data-loader/portal/users.js:43]: Caroline is fine with this email address

File: e2e-tests/portal/cypress/integration/journeys/dashboard/dashboard-deals/dashboard-deals-create-deal.spec.js (hunk 1 above)
Comment (ttbarnes) [reply e2e-tests/portal/cypress/integration/journeys/dashboard/dashboard-deals/dashboard-deals-create-deal.spec.js:29]: Ignore - ideally we'd have middleware to cover all BSS routes. This is not a small job and we can't do right now - as it is now is ok for V1.

**Review by IgnatG — DISMISSED** [review]
LGTM

**Review by ttbarnes — DISMISSED** [review]

**Review by ttbarnes — APPROVED** [review]

**Review by IgnatG — APPROVED** [review]

File: dtfs-central-api/src/v1/controllers/tfm/deal/tfm-update-deal.controller.js [hunk 5]
```diff
@@ -68,12 +68,23 @@ const updateDeal = async (dealId, dealChanges, existingDeal) => {
   const existingDealActivities = (existingDeal.tfm && existingDeal.tfm.activities);
   const tfmUpdateHasActivities = (tfmUpdate.activities
                                   && Object.keys(tfmUpdate.activities).length > 0);
-
+  /**
+   * ACBS activities update are an array whereas TFM activities update is an object
```
Comment (abhi-markan) [inline dtfs-central-api/src/v1/controllers/tfm/deal/tfm-update-deal.controller.js:72]: Small - update `is` an array whereas TFM `activity`

**Review by abhi-markan — CHANGES_REQUESTED** [review]

File: dtfs-central-api/src/v1/controllers/tfm/deal/tfm-update-deal.controller.js (hunk 5 above)
Comment (Zainzzkk) [reply dtfs-central-api/src/v1/controllers/tfm/deal/tfm-update-deal.controller.js]: done

**Review by abhi-markan — APPROVED** [review]

**Pull request merged**